test/odbc/psqlodbc/sql_variant.cpp

```diff
@@ -0,0 +1,614 @@
+    }
+    // <> operator
+    SqlVariant_OperationsQueryBuilder(i, i, "<>", bbf_operations_query, pg_operations_query);
+  }
+}
+
+// Helper function for PSQL_DataTypes_Sql_Variant.Comparison_Operators to validate that all comparisons
+//  the @is_all_true indiciates if we expect all values to be true, or all values to be false
+void SqlVariant_VerifyOperationsQuery(ServerType server_type, const bool &is_all_true, vector<string> &operations_query) {
+  const int BUFFER_LEN = 255;;
+  const string EXPECTED_RESULTS = is_all_true ? "1" : "0";
+  const SQLLEN EXPECTED_LEN = 1;
+  char data[BUFFER_LEN];
+  RETCODE rcode;
+  SQLLEN data_len;
+
+  OdbcHandler odbcHandler(Drivers::GetDriver(server_type));
+  odbcHandler.Connect(true);
+
+  const vector<tuple<int, int, SQLPOINTER, int, SQLLEN*>> constraints_bind_columns = {
+    {1, SQL_C_CHAR, data, BUFFER_LEN, &data_len}
+  };
+
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(constraints_bind_columns));
+
+  for (int i = 0; i < operations_query.size(); i++) {
+    odbcHandler.ExecQuery(operations_query[i]);
+
+    rcode = SQLFetch(odbcHandler.GetStatementHandle());
+    EXPECT_EQ(rcode, SQL_SUCCESS);
+    EXPECT_EQ(string(data), EXPECTED_RESULTS);
+    EXPECT_EQ(data_len, EXPECTED_LEN);
+
+    rcode = SQLFetch(odbcHandler.GetStatementHandle());
+    EXPECT_EQ(rcode, SQL_NO_DATA);
+    odbcHandler.CloseStmt();
+  }
+
+  return;
+}
+
+TEST_F(PSQL_DataTypes_Sql_Variant, Table_Creation) {
+  createTable(ServerType::MSSQL, BBF_TABLE_NAME, TABLE_COLUMNS);
+
+  const vector<int> BBF_LENGTH_EXPECTED = {10, 8000};
+  const vector<int> BBF_PRECISION_EXPECTED = {10, 8000};
+  const vector<int> BBF_SCALE_EXPECTED = {0, 0};
+  const vector<string> BBF_NAME_EXPECTED = {"int", "sql_variant"};
+
+  testCommonColumnAttributes(ServerType::MSSQL, BBF_TABLE_NAME, 
+    TABLE_COLUMNS.size(), COL1_NAME, 
+    BBF_LENGTH_EXPECTED, BBF_PRECISION_EXPECTED, 
+    BBF_SCALE_EXPECTED, BBF_NAME_EXPECTED);
+
+  const vector<int> PG_LENGTH_EXPECTED = {4, 255};
+  const vector<int> PG_PRECISION_EXPECTED = {0, 0};
+  const vector<int> PG_SCALE_EXPECTED = {0, 0};
+  const vector<string> PG_NAME_EXPECTED = {"int4", "unknown"};
+
+  testCommonColumnAttributes(ServerType::PSQL, PG_TABLE_NAME, 
+    TABLE_COLUMNS.size(), COL1_NAME, 
+    PG_LENGTH_EXPECTED, PG_PRECISION_EXPECTED, 
+    PG_SCALE_EXPECTED, PG_NAME_EXPECTED);
+
+  dropObject(ServerType::PSQL, "TABLE", PG_TABLE_NAME);
+  dropObject(ServerType::MSSQL, "TABLE", BBF_TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Sql_Variant, Insertion_Success) {
+  // Tuple of vectors that contains the following values in the corresponding spots
+  //   1. Inserted_values (same as expected from bbf)
+  //   2. Expected values for pg (empty string if the same as bbf)
+  //   3. Datatype
+  //   4. Single-quote-wrapped (needs single quotes around the data inserted)
+  const vector<tuple<string, string, string, bool>> INSERTION_INFO = {
```

Comment:
This seems to be used a lot, and it's a constant so I don't believe it gets modified. Can we move this to the top of the test file as a global variable?

Reply:
Agreed; however, there's a few places the values of this is not changed and a few times where one row gets commented out because of a failed test. I'll see what i can do to group the common ones together to a global

Reply:
Done, I separated it into 2 global variables since they could be seperated into 2 groups (one for constraints, and one for insertion and view).